### ISSUE TITLE
4k 60fps hdr video has a lot of dropped frames on full screen.

### ISSUE THREAD

**Issue by armodim** [mpv-player/mpv#8621]
- mpv version: 0.33.0-4
- Linux Distribution and Version: Arch Linux
- Source of the mpv binary: Arch packages repository
- Window Manager and version: sway/wayland
- CPU: i9-9900k.
- monitor UHD, 60Hz.
- vainfo: VA-API version: 1.10 (libva 2.10.0)
- vainfo: Driver version: Intel i965 driver for Intel(R) Coffee Lake - 2.4.1

### Reproduction steps
Play 4k 60fps hdr video with --hwdec=vaapi.

(With --no-config option there are twice more dropped frames and the CPU usage goes up to 50-60%.)

### Expected behavior
No stuttering or choppy. 

### Actual behavior
Video is choppy and has a lot of dropped frames. But playing video in vlc has no any stuttering (there is another issue though - colors are very poor). 
Exiting full screen (downscale) or putting mpv to a FullHD monitor make video smooth.

### Log file
Screenshot of Shift+i: https://0x0.st/-POo.png
Screenshot of htop while playing video: https://0x0.st/-POZ.png
Log-file: https://0x0.st/-PO8.txt

### Sample files
For the tests I play downloaded file from this youtube video:
https://www.youtube.com/watch?v=LXb3EKWsInQ

**Comment by Dudemanguy** [comment]
>Mesa Intel(R) UHD Graphics 630

Isn't your GPU just underpowered? I don't know what settings you used for vlc but on my machine, the opengl output on vlc for this video is extremely choppy and drops a ton of frames for me. mpv works fine with the default (things get dicey if I start to use more advanced settings though).

Probably on your hardware, software scalers would for fine (like wlshm) but the colors will be awful since there is no tonemapping. Likely whatever you were doing with vlc was also using a software scaler (hence the poor colors).

Basically, vo=gpu is well, gpu-bound.

**Comment by philipl** [comment]
Yes, you can't do it. UHD 630 is only capable of doing 4k 24fps.

**Comment by armodim** [comment]
> Yes, you can't do it. UHD 630 is only capable of doing 4k 24fps.
Yes if you use HDMI 1.4. But via DisplayPort it is ok. Besides it vlc playing it smooth, as i said.
I guess the key of the problem is hdr. Have a look into my screenshot at the topic ( https://0x0.st/-POo.png ). There is Pixel Format:"**p010**" with Primaries: **bt.2020**. 
And 4k 60fps NO hdr video (for example https://www.youtube.com/watch?v=V5Fsj_sCKdg ) is playing smooth. Here the screenshot of it: https://0x0.st/-P4O.png
As you can see, there is no drops. BUT the Pixel Format: **nv12**, Primaries: **bt.709**.

**Comment by philipl** [comment]
Sorr, let me be more specific. My point was that there isn't enough GPU power to do 4k60 HDR with a UHD 630. As @Dudemanguy discussed, there isn't enough GPU power to do the tonemapping.

**Comment by armodim** [comment]
> Isn't your GPU just underpowered? 
CPU is not loaded at all. 12-20%. And it's cold, the fans keep silence as well. I could say there is no any difference against idle system.

**Comment by Dudemanguy** [comment]
>CPU is not loaded at all

I said **GPU**. You need a better GPU if you want to do tonemapping at 4k60.

**Comment by armodim** [comment]
> > CPU is not loaded at all
> 
> I said **GPU**. 

GPU integrated into a CPU. So if the gpu is overloaded, htop would show this. Init?

**Comment by Dudemanguy** [comment]
>So if the gpu is overloaded, htop would show this.

I'm fairly certain it wouldn't. I'm pretty sure the iGPU has no effect on the CPU aside from perhaps like RAM/memory usage. You would need some other dedicated tool if you want to measure it. Anyways, closing this since it is a hardware issue (not powerful enough for tonemapping).

**Comment by CounterPillow** [comment]
> GPU integrated into a CPU. So if the gpu is overloaded, htop would show this. Init?

no, that's what intel_gpu_top is for. Hilarious logic, though.

**Comment by armodim** [comment]
> > GPU integrated into a CPU. So if the gpu is overloaded, htop would show this. Init?
> 
> no, that's what intel_gpu_top is for. Hilarious logic, though.

Didn't now about the tool. Thanks!
https://0x0.st/-P4h.png

**Comment by bedrin** [comment]
Try `hdr-compute-peak=no`

**Comment by zer0-x** [comment]
I had the same problem with 4K, my GPU was not loaded as is my CPU. I was able to fix it switching to the Vulkan API.

```
--gpu-api=vulkan
```

> **Note** : Vulkan might cause some problems for some people depending on your drivers and your hardware.
